### DISCUSSION TITLE
exclude file upload from form validation?

### DISCUSSION THREAD

**Discussion by rothskeller** [unpoly/unpoly#448] [Q&A]
Is there a way to tell unpoly not to upload files during form validation?  I have a form with a number of fields, including one file upload field.  The files uploaded through this form tend to be quite large, and are not needed for validation; they are only needed on actual submission.  Uploading them during validation makes validation unacceptably slow.

**Comment by triskweline** [comment]
You might be able to filter the parameter list before the request is sent:

```js
function isBinary(value) {
  return value instanceof Blob
}

up.on('up:form:validate', ({ renderOptions }) => {
  let params = renderOptions.params
  let entriesWithoutFiles = params.toArray().filter(({ name, value }) => !isBinary(values))
  renderOptions.params = entriesWithoutFiles
})
```

I haven't tested the code above.

**Comment by hpvd** [comment]
Since there are often -beside fields- uploads within forms, this is an interesting topic.
Is there already a way or an idea to give uploads some kind of lower priority so users can e.g.
- receive validations,
- do their corrections and
- even send for revalidation

while upload is in progress?

It's not only relevant for huge uploads, but also small uploads may block your traffic if you are on a slow mobile connection e.g. within a crowded train...
